### PR TITLE
[dynamo] log all graph breaks to graph_breaks logging artifact

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137244

We were previously not logging all graph breaks (e.g. data dependent jumps) to the graph_breaks logging artifact.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: test/dynamo/test_misc.py [hunk 1]
```diff
@@ -5795,7 +5795,9 @@ def g2(a, b):
             return a + b + c
 
         def count_graph_break_msgs(msgs):
-            return sum(msg.find("Graph break") != -1 for msg in msgs)
+            return sum(
+                "Graph break" in msg and "details suppressed" not in msg for msg in msgs
```
Comment (ezyang) [inline test/dynamo/test_misc.py:5799]: ?

**Comment by mlazos** [comment]
At a high level, why do we have separate graph break handling in those two places? 

What I'm thinking is that could we put the log graph break logic in `compile_subgraph` and check whether the compile reason is graph break? That would've prevented us from missing these other `compile_subgraph` calls.

File: test/dynamo/test_misc.py (hunk 1 above)
Comment (williamwen42) [reply test/dynamo/test_misc.py:5799]: Filtering out details suppressed graph break messages (to comply with removing duplicate detailed graph break messages)

**Comment by williamwen42** [comment]
@mlazos good point, let me try doing that.

File: test/dynamo/test_misc.py (hunk 1 above)
Comment (williamwen42) [reply test/dynamo/test_misc.py:5799]: Simplified this check.

**Comment by williamwen42** [comment]
@mlazos I'm going to merge this PR for now and refactor in a followup PR.

**Pull request closed without merging**